src/Evolution/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -14,21 +14,111 @@
 #include "Domain/InitialElementIds.hpp"
 #include "Domain/OptionTags.hpp"
 #include "Domain/Tags.hpp"
+#include "Evolution/Protocols.hpp"
+#include "Evolution/Tags.hpp"
+#include "IO/Importers/VolumeDataReader.hpp"
+#include "IO/Importers/VolumeDataReaderActions.hpp"
 #include "Parallel/ConstGlobalCache.hpp"
 #include "Parallel/Info.hpp"
 #include "Parallel/ParallelComponentHelpers.hpp"
+#include "ParallelAlgorithms/Actions/SetData.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
+/*!
+ * \brief Don't try to import initial data from a data file
+ *
+ * \see `DgElementArray`
+ */
+struct ImportNoInitialData {};
+
+/*!
+ * \brief Import numeric initial data in the `ImportPhase`
+ *
+ * Requires the `InitialData` conforms to
+ * `evolution::protocols::NumericInitialData`. The `InitialData::import_fields`
+ * will be imported in the `ImportPhase` from the data file that is specified by
+ * the options in the `evolution::OptionTags::NumericInitialData` group.
+ *
+ * \see `DgElementArray`
+ */
+template <typename PhaseType, PhaseType ImportPhase, typename InitialData>
+struct ImportNumericInitialData {
+  static_assert(
+      tt::conforms_to_v<InitialData, evolution::protocols::NumericInitialData>);
+};
+
+namespace DgElementArray_detail {
+
+template <typename DgElementArray, typename InitialData,
+          typename ImportFields = typename InitialData::import_fields>
+using read_element_data_action = importers::ThreadedActions::ReadVolumeData<
+    evolution::OptionTags::NumericInitialData, ImportFields,
+    ::Actions::SetData<ImportFields>, DgElementArray>;
+
+template <typename DgElementArray, typename ImportInitialData>
+struct import_numeric_data_cache_tags {
+  using type = tmpl::list<>;
+};
+
+template <typename DgElementArray, typename PhaseType, PhaseType ImportPhase,
+          typename InitialData>
+struct import_numeric_data_cache_tags<
+    DgElementArray,
+    ImportNumericInitialData<PhaseType, ImportPhase, InitialData>> {
+  using type =
+      typename read_element_data_action<DgElementArray,
+                                        InitialData>::const_global_cache_tags;
+};
+
+template <typename DgElementArray, typename ImportInitialData>
+struct try_import_data {
+  template <typename Metavariables>
+  static void apply(const typename Metavariables::Phase /*next_phase*/,
+                    Parallel::CProxy_ConstGlobalCache<
+                        Metavariables>& /*global_cache*/) noexcept {}
+};
+
+template <typename DgElementArray, typename PhaseType, PhaseType ImportPhase,
+          typename InitialData>
+struct try_import_data<
+    DgElementArray,
+    ImportNumericInitialData<PhaseType, ImportPhase, InitialData>> {
+  template <typename Metavariables>
+  static void apply(
+      const typename Metavariables::Phase next_phase,
```

Comment:
Since you have `Metavariables` in this function, is it not possible to instead check from the metavariables if numerical data is being imported? Then you can use `if constexpr` instead of the specialization.